### ISSUE TITLE
Syntax question: to `let` or not to `let`

### ISSUE THREAD

**Issue by nikomatsakis** [dada-lang/dada#217]
In the current dada syntax, if you use `=` you declare a new variable:

```
x = 22
```

and if you use `:=`, you can reassign an existing place:

```
x := 23
y.field := 23
```

I was going for an extremely "lightweight" syntax reminiscent of Python, but to be honest, I'm not sure I like it. I am thinking maybe it's better to follow Rust (and, I realize, Java/TypeScript?) precedent here and use `let`:

```
let x = 22
```

and then just use `=` for reassignment:

```
x = 23
y.field = 23
```

Thoughts?

**Comment by nikomatsakis** [comment]
This probably makes parsing easier too, as you would presumably write

```
let x: Type
```

and 

```
let atomic x
```

Right now we need a bit of goofy lookahead to handle these cases.

**Comment by vemoo** [comment]
I prefer having `let`, it's easier to see where new bindings are being introduced.

**Comment by nikomatsakis** [comment]
@vemoo want to try implementing it?

**Comment by vemoo** [comment]
sure

**Comment by brson** [comment]
I also prefer keywords to introduce variables. I think it's because it's a strong visual indicator that new state is being created.

**Comment by vemoo** [comment]
this was implemented in #221
